common/dynamicconfig/collection.go

```diff
@@ -105,14 +105,12 @@ func NewCollection(client Client, logger log.Logger) *Collection {
 	}
 }
 
-func (c *Collection) logError(key Key, err error) {
+func (c *Collection) throttleLog() bool {
 	// TODO: This is a lot of unnecessary contention with little benefit. Consider using
 	// https://github.com/cespare/percpu here.
 	errCount := atomic.AddInt64(&c.errCount, 1)
-	if errCount%errCountLogThreshold == 0 {
-		// log only every 'x' errors to reduce mem allocs and to avoid log noise
-		c.logger.Debug("No such key in dynamic config, using default", tag.Key(key.String()), tag.Error(err))
-	}
+	// log only the first x errors and then x after that to reduce log noise
+	return errCount < errCountLogThreshold || errCount%errCountLogThreshold == 0
```

Comment:
The idea here is to get all the distinct errors upfront instead of just the first error that happens on startup, right?